tiamat/src/commonMain/kotlin/com/composegears/tiamat/Utils.kt

```diff
@@ -1,8 +1,12 @@
 package com.composegears.tiamat
 
 /**
- * We can not call T::class in @Composable functions,
- *
- * workaround is to call it outside of @Composable via regular inline fun
+ * @return true if all elements match the given predicate or collection is empty
  */
-inline fun <reified T : Any> className(): String = T::class.qualifiedName!!
\ No newline at end of file
+internal fun <T> Iterable<T>.allIndexed(predicate: (Int, T) -> Boolean): Boolean {
+    if (this is Collection && isEmpty()) return true
```

Comment:
the whole extension looks the same as `all{}` 

<img width="719" alt="image" src="https://github.com/ComposeGears/Tiamat/assets/16294951/dbd2b570-a8c9-48d8-9e4e-778f83a4662f">

what about
```kotlin
internal fun <T> Iterable<T>.allIndexed(predicate: (Int, T) -> Boolean): Boolean {
    return withIndex().all { predicate(it.index, it.value) }
}
```

Reply:
withIndex() will create a wrapper of iterator and do alot extra work to calc index, 
forEachIndexed under the hood is just `int++` + `simple iteration` - this is more optimized and faster